### PR TITLE
Migrating to CareUI components in ResourceDetailsUpdate.tsx component

### LINKED ISSUE #4994: Replace Material UI components in `src/Components/Resource/ResourceDetailsUpdate.tsx`:
```tsx
23: import { CircularProgress } from "@material-ui/core";\n32: } from "@material-ui/core";
```

### Tasks

Ensure all the below tasks are accomplished if applicable to the file being modified in the PR.

- [ ] Migrate all Material UI components to use new Care UI / custom components. (ref: [🛠️ Tailwinding Care](https://github.com/coronasafe/care_fe/issues/3742))
- [ ] Migrate all Material UI icons to use [🏥 **Care Icons**](https://github.com/coronasafe/care_fe/pull/4071).
- [ ] Migrate `PageTitle` to `Page` component if any. (See [example](https://github.com/coronasafe/care_fe/pull/4796/files#diff-8e39f154ccf13c9dedd0f968d21a0abbde450e88df9724a4e7d4264d487a7846))

For more information see the epic issue that tracks this issue.

### PR DESCRIPTION
**Pull request by MayaSatishRao**
### <samp>🤖 Generated by Copilot at 5855b3a</samp>

This pull request improves the user interface and experience of the consultation and resource management features, and fixes a bug in the `AutoCompleteAsync` component. It adds a function to show customized notification messages in `ConsultationForm.tsx`, refactors `ResourceDetailsUpdate.tsx` to use the CAREUI library components, and clears the query state and disables the browser's autocomplete in `AutoCompleteAsync.tsx`.

## Proposed Changes

- Fixes #4994 
- migrated from MaterialUI components to CareUI components

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

### CODE REVIEW

**Comment by MayaSatishRao** [comment]
@rithviknishad Could u please verify if all the materialUI components are migrated?

File: src/Components/Resource/ResourceDetailsUpdate.tsx [hunk 1]
```diff
@@ -230,155 +225,127 @@ export const ResourceDetailsUpdate = (props: resourceProps) => {
...
+                name="emergency"
+                onChange={handleChange}
+                label={"Is this an emergency?"}
+                options={[true, false]}
+                optionDisplay={(o) => (o ? "Yes" : "No")}
+                optionValue={(o) => String(o)}
+                value={state.form.emergency}
+              />
+
+              <FieldErrorText error={state.errors.emergency} />
```
Comment (rithviknishad) [inline src/Components/Resource/ResourceDetailsUpdate.tsx]: do not use FieldLabel or FieldErrorText on form field components. Use it's label or error props instead.

File: src/Components/Resource/ResourceDetailsUpdate.tsx [hunk 2]
```diff
@@ -230,155 +225,127 @@ export const ResourceDetailsUpdate = (props: resourceProps) => {
...
-                  selected={state.form.approving_facility_object}
-                  setSelected={(obj) =>
-                    setFacility(obj, "approving_facility_object")
-                  }
-                  errors={state.errors.approving_facility}
-                />
+        <Card className="w-full flex flex-col">
+          <div className="grid gap-4 grid-cols-1 md:grid-cols-2">
+            <div className="md:col-span-1">
+              <FieldLabel>Status</FieldLabel>
```
Comment (rithviknishad) [inline src/Components/Resource/ResourceDetailsUpdate.tsx]: same here too

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Comment by rithviknishad** [comment]
Also update your PR title to something meaningful.

**Review by rithviknishad — CHANGES_REQUESTED** [review]
You've also not migrated PageTitle component

**Comment by MayaSatishRao** [comment]
I have made the requested changes @rithviknishad

**Review by rithviknishad — APPROVED** [review]
LGTM. You could have passed the classNames directly to the Form field components instead of wrapping it in a div and specifying it there. Code would have looked cleaner!

**Review by rithviknishad — CHANGES_REQUESTED** [review]
@MayaSatishRao use the new `UserAutocompleteFormField` by merging latest develop

**Comment by rithviknishad** [comment]
@MayaSatishRao any updates on this?

**Comment by MayaSatishRao** [comment]
@rithviknishad I have made the changes.

**Review by rithviknishad — CHANGES_REQUESTED** [review]
Are you sure you have replaced all the MUI components?

**Comment by rithviknishad** [comment]
There shouldn't be any imports of material ui

**Comment by MayaSatishRao** [comment]
@rithviknishad  i have made the changes. I am still using FieldLabel for FacilitySelect because it does not have label prop. Other than this all other components are from CareUI

**Comment by nihal467** [comment]
LGTM

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**